### PR TITLE
Fix bad SSL connections.

### LINKED ISSUE #528: V 1.3.6 Error: There was an error communicating with your Plex Server. Please check your settings
Issue:
Enabling Secure Connections to "**Require**" on my PMS results in PlexPy homescreen giving the following error “There was an error communicating with your Plex Server. Please check your settings” for the Activity and Recently added sections.

Enabling Secure Connections to "**Preferred**" on my PMS results in PlexPy operating normally and able to display currently playing items plus it can retrieve and display the recently added items.

Network Environment:
192.168.1.0/24 based network
PMS runs on my Unraid server at 192.168.1.21:32400      PMS Version 0.9.15.2
PlexPy running on my PC at 192.168.1.21:8181        PlexPy Version 1.3.6
PlexPy running on my Raspberry Pi at 192.168.1.157:8989        PlexPy Version 1.3.6.  
I also tried the default port of 8181 with the same results.

This issue occurs on both my PC and the Rasberry Pi. I have tried enabling SSL and telling PlexPy it is on a remote network in multiple combinations.  No combination worked.
SSL=on   Remote Network =off
SSL=on   Remote Network =on
SSL=off   Remote Network =on

My normal environment is for me to run secure connections.  My local Fire TV boxes, local web based playback and remote access works correctly.  That’s not to say it could not be something local to my environment, just saying other devices work are working.

I was installing PlexPy for the first time on my PC when I encountered the error.  Nothing I did would clear the error.  I was able to get the PC based PlexPy to access my Plex log files but I still got the error.  

I decided to try and install it onto my Raspberry Pi where I ultimately want it to reside as a dedicated server running 24/7.  Got the same “There was an error communicating with your Plex Server. Please check your settings” error.  Somewhere in my trouble shooting I changed the requirement in Plex to disable SSL and everything started working in both installs of PlexPy

As a side note if it comes up with another user, to get the Plex log access working, I had to mount the Unraid share where the log is stored to a local mount point.  I could then map the path within PlexPy.

Any guidance on the source of the issue and how to correct it would be appreciated.  Running without SSL on my server is not an option.  Again turning off SSL on the PMS server and everything works in PlexPy.  Turn on SSL required in PMS and PlexPY cannot access the server for data.

**Beginning of log file after reboot with Force SSL button checked.**
2016-02-07 03:54:29 WARNING PlexPy Pmsconnect :: Unable to parse XML for get_sessions: 'NoneType' object has no attribute 'getElementsByTagName'.
2016-02-07 03:54:29 WARNING Unable to retrieve data for get_recently_added.
2016-02-07 03:54:29 WARNING Failed to access uri endpoint /status/sessions with error [Errno -5] No address associated with hostname
2016-02-07 03:54:29 WARNING PlexPy Pmsconnect :: Unable to parse XML for get_recently_added: 'NoneType' object has no attribute 'getElementsByTagName'.
2016-02-07 03:54:29 WARNING PlexPy Pmsconnect :: Unable to parse XML for get_sessions: 'NoneType' object has no attribute 'getElementsByTagName'.
2016-02-07 03:54:29 WARNING Failed to access uri endpoint /library/recentlyAdded?X-Plex-Container-Start=0&X-Plex-Container-Size=10 with error [Errno -5] No address associated with hostname
2016-02-07 03:54:29 WARNING Failed to access uri endpoint /status/sessions with error [Errno -5] No address associated with hostname
2016-02-07 03:54:15 INFO    Scheduled background task: Refresh libraries list
2016-02-07 03:54:15 INFO    Scheduled background task: Refresh users list
2016-02-07 03:54:15 INFO    Scheduled background task: Check for active sessions
2016-02-07 03:54:15 INFO    Scheduled background task: Check for server response
2016-02-07 03:54:15 INFO    Scheduled background task: Refresh Plex Server Name
2016-02-07 03:54:15 INFO    Scheduled background task: Refresh Plex Server URLs
2016-02-07 03:54:15 INFO    Scheduled background task: Check GitHub for updates
2016-02-07 03:54:14 INFO    Starting PlexPy web server on http://0.0.0.0:8989/
2016-02-07 03:54:14 WARNING PlexPy Pmsconnect :: Unable to refresh libraries list.
2016-02-07 03:54:14 WARNING PlexPy Pmsconnect :: Unable to parse XML for get_libraries_list: 'NoneType' object has no attribute 'getElementsByTagName'.
2016-02-07 03:54:14 WARNING Failed to access uri endpoint /library/sections with error [Errno -5] No address associated with hostname
2016-02-07 03:54:14 INFO    PlexPy Pmsconnect :: Requesting libraries list refresh...
2016-02-07 03:54:14 INFO    PlexPy PlexTV :: Users list refreshed.
2016-02-07 03:54:13 INFO    PlexPy PlexTV :: Requesting users list refresh...
2016-02-07 03:54:13 WARNING PlexPy Pmsconnect :: Unable to parse XML for get_server_list: 'NoneType' object has no attribute 'getElementsByTagName'.
2016-02-07 03:54:13 WARNING Failed to access uri endpoint /servers with error [Errno -5] No address associated with hostname
2016-02-07 03:54:13 WARNING PlexPy Pmsconnect :: Unable to parse XML for get_local_server_name: 'NoneType' object has no attribute 'getElementsByTagName'.
2016-02-07 03:54:13 WARNING Failed to access uri endpoint /:/prefs with error [Errno -5] No address associated with hostname
2016-02-07 03:54:12 INFO    PlexPy Pmsconnect :: Requesting name from server...
2016-02-07 03:54:12 INFO    PlexPy PlexTV :: Server URL retrieved.
2016-02-07 03:54:12 INFO    Writing configuration to file
2016-02-07 03:54:11 INFO    Writing configuration to file
2016-02-07 03:54:11 INFO    PlexPy PlexTV :: Requesting URLs for server...
2016-02-07 03:54:11 INFO    PlexPy is up to date
2016-02-07 03:54:11 DEBUG   Latest version is 49fb4540a26e56af4e505644f6e549622da9d80d
2016-02-07 03:54:10 DEBUG   Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/commits/master
2016-02-07 03:54:10 INFO    Retrieving latest version information from GitHub
2016-02-07 03:54:10 DEBUG   Git output: master
2016-02-07 03:54:10 DEBUG   Trying to execute: "git rev-parse --abbrev-ref HEAD" with shell in /opt/plexpy
2016-02-07 03:54:10 DEBUG   Git output: 49fb4540a26e56af4e505644f6e549622da9d80d
2016-02-07 03:54:10 DEBUG   Trying to execute: "git rev-parse HEAD" with shell in /opt/plexpy
2016-02-07 03:54:10 INFO    Checking to see if the database has all tables

### PR DESCRIPTION
**Pull request by drzoidberg33**
Fixes #528.

### CODE REVIEW

**Comment by drzoidberg33** [comment]
@Hellowlol and @JonnyWong16 Let me know if you see something off here.

File: plexpy/config.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def bool_int(value):
     'PMS_NAME': (unicode, 'PMS', ''),
     'PMS_PORT': (int, 'PMS', 32400),
     'PMS_TOKEN': (str, 'PMS', ''),
-    'PMS_SSL': (int, 'General', 0),
+    'PMS_SSL': (int, 'PMS', 0),
```
Comment (JonnyWong16) [inline plexpy/config.py]: If `pms_ssl` is set to `1` under `[General]`, then it doesn't carry over to `[PMS]`.

**Comment by JonnyWong16** [comment]
I still don't think you can access `/identity` over HTTP if secure connections is set to "Required".

```
07-Feb-2016 09:53:10 - WARNING :: CP Server Thread-11 : Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? ''
07-Feb-2016 09:53:10 - WARNING :: CP Server Thread-11 : Unable to retrieve the PMS identifier.
```

---

**TL;DR: This P/R.**
- "Preferred" and Use SSL "unchecked" works.
- "Preferred" and Use SSL "checked" grabs the wrong `pms_url` for non-remote.
- "Required" and Use SSL "unchecked" doesn't verify the server.
- "Required" and Use SSL "checked" doesn't verify the server.

**TL;DR: Before this P/R.**
- "Preferred" and Use SSL "unchecked" works.
- "Preferred" and Use SSL "checked" grabs the wrong `pms_url` for non-remote.
- "Required" and Use SSL "unchecked" works (it shouldn't connect)
- "Required" and Use SSL "checked" grabs the wrong `pms_url` for non-remote

---

Something is definitely broken. I have 3 possible connections, including a "Custom server access URLs" set on my Plex server. (connection 2 below).

```
<Connection protocol="https" address="192.168.xxx.xxx" port="32400" uri="https://192-168-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:32400" local="1"/>
<Connection protocol="https" address="xxxxxxxxxx.xxxxxxxxxx.com" port="xxxxx" uri="https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx" local="1"/>
<Connection protocol="https" address="174.xxx.xxx.xxx" port="xxxxx" uri="https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx" local="0"/>
```

This is from testing every combination. The "Connects" column means PlexPy can access the uri endpoints. The "PMS URL" is the one grabbed automatically by `plextv.get_real_pms_url()`.

One case of note is Secure Settings "Preferred" and Use SSL "Checked", it verifies, but always grabs my custom URL, which means it cannot connect.

| Secure Settings | Use SSL | Remote Server | IP/Hostname | Verifies | Connects | PMS URL |
| --- | --- | --- | --- | --- | --- | --- |
| Preferred | Unchecked | Unchecked | Local IP | Yes | Yes | http://192.168.xxx.xxx:32400 |
| Preferred | Unchecked | Unchecked | Custom URL | Yes | Yes | http://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Unchecked | Unchecked | Remote IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Local IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Custom URL | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Remote IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Checked | Unchecked | Local IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Unchecked | Custom URL | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Unchecked | Remote IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Checked | Local IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Preferred | Checked | Checked | Custom URL | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Preferred | Checked | Checked | Remote IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Required | Unchecked | Unchecked | Local IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Unchecked | Unchecked | Custom URL | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Unchecked | Unchecked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Unchecked | Checked | Local IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Unchecked | Checked | Custom URL | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Unchecked | Checked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Unchecked | Local IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Unchecked | Custom URL | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Unchecked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Checked | Local IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Checked | Custom URL | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Checked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |

---

This is before this P/R

| Secure Settings | Use SSL | Remote Server | IP/Hostname | Verifies | Connects | PMS URL |
| --- | --- | --- | --- | --- | --- | --- |
| Preferred | Unchecked | Unchecked | Local IP | Yes | Yes | http://192.168.xxx.xxx:32400 |
| Preferred | Unchecked | Unchecked | Custom URL | Yes | Yes | http://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Unchecked | Unchecked | Remote IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Local IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Custom URL | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Remote IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Checked | Unchecked | Local IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Unchecked | Custom URL | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Unchecked | Remote IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Checked | Local IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Preferred | Checked | Checked | Custom URL | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Preferred | Checked | Checked | Remote IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Required | Unchecked | Unchecked | Local IP | Yes | No | http://192.168.xxx.xxx:32400 |
| Required | Unchecked | Unchecked | Custom URL | Yes | No | http://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Required | Unchecked | Unchecked | Remote IP | Yes | No | http://174.xxx.xxx.xxx:xxxxx |
| Required | Unchecked | Checked | Local IP | Yes | No | http://174.xxx.xxx.xxx:xxxxx |
| Required | Unchecked | Checked | Custom URL | Yes | No | http://174.xxx.xxx.xxx:xxxxx |
| Required | Unchecked | Checked | Remote IP | Yes | No | http://174.xxx.xxx.xxx:xxxxx |
| Required | Checked | Unchecked | Local IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Required | Checked | Unchecked | Custom URL | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Required | Checked | Unchecked | Remote IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Required | Checked | Checked | Local IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Required | Checked | Checked | Custom URL | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Required | Checked | Checked | Remote IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |

**Comment by drzoidberg33** [comment]
Thanks Jonny! Yes, I see `identity` doesn't work on remote servers.

I'll fix that.

File: plexpy/webserve.py [hunk 2]
```diff
@@ -1187,6 +1187,11 @@ def configUpdate(self, **kwargs):
             (kwargs['monitor_remote_access'] != plexpy.CONFIG.MONITOR_REMOTE_ACCESS):
             reschedule = True
 
+        # If we change the SSL setting for PMS, make sure we grab the new url.
+        if 'pms_ssl' in kwargs and \
+            (kwargs['pms_ssl'] != plexpy.CONFIG.PMS_SSL):
```
Comment (JonnyWong16) [inline plexpy/webserve.py:1192]: Also grab new url when changing `pms_is_remote` setting.

Comment (drzoidberg33) [inline plexpy/webserve.py:1192]: Thanks, done.

**Comment by drzoidberg33** [comment]
@JonnyWong16 You mind testing again?

File: plexpy/config.py (hunk 1 above)
Comment (drzoidberg33) [inline plexpy/config.py]: Ye, this is a silly bug. I'll probably just put it back then.

**Comment by JonnyWong16** [comment]
Still have the problem where `plextv.get_real_pms_url()` grabs my custom URL even when i use the IP address.

And now "Required" cannot verify my remote IP address.

**Comment by drzoidberg33** [comment]
> Still have the problem where plextv.get_real_pms_url() grabs my custom URL even when i use the IP address.

Can you tell me what steps to reproduce? I cannot figure this out, looks good on my side if I use the IP.

> And "Required" cannot verify my remote IP address.

If you're trying to use a remote address when there is a private address available it won't be returned from the `discover` method due to my change at https://github.com/drzoidberg33/plexpy/pull/529/files#diff-cb42729bf12c3bdce109028c021370b7R501

Not sure why you would want to do this in the first place?

Am I understanding correctly here?

**Comment by Hellowlol** [comment]
I havnt followed the thread so a really have nothing to add, but regarding the diff you linked. My point with the discovery feature was to find all urls and let the user pick and choose. I dont understand other limitations was added. But ill look into it :P

**Comment by drzoidberg33** [comment]
> I havnt followed the thread so a really have nothing to add, but regarding the diff you linked. My point with the discovery feature was to find all urls and let the user pick and choose. I dont understand other limitations was added. But ill look into it :P

The problem there was that it was also returning local IPs for remote servers. This is actually the main fix of this PR as it was causing problems if you had another server with the same local IP (it would possibly assign the incorrect machineID).

**Comment by drzoidberg33** [comment]
To use your checklist:
- "Preferred" and Use SSL "unchecked" works. **Works**
- "Preferred" and Use SSL "checked" grabs the wrong pms_url for non-remote. **Works**
- "Required" and Use SSL "unchecked" doesn't verify the server. **Works**
- "Required" and Use SSL "checked" doesn't verify the server. **Works**

**Comment by JonnyWong16** [comment]
> Can you tell me what steps to reproduce? I cannot figure this out, looks good on my side if I use the IP.
1. In the Plex server settings > Network > Custom server access URLs, I have `https://mynetwork.mydomain.com:port`
2. Secure connections is set to "Preferred".
3. I have remote access enabled.
4. On `https://plex.tv/api/resources?includeHttps=1`, I have 3 connections listed for my server.
   
   i. Local IP
   ii. Custom URL
   iii. Remote IP
   
   ```
   <Connection protocol="https" address="192.168.xxx.xxx" port="32400" uri="https://192-168-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:32400" local="1"/>
   <Connection protocol="https" address="xxxxxxxxxx.xxxxxxxxxx.com" port="xxxxx" uri="https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx" local="1"/>
   <Connection protocol="https" address="174.xxx.xxx.xxx" port="xxxxx" uri="https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx" local="0"/>
   ```
5. In PlexPy > Settings > Plex Media Server, I enter my local IP address **OR** my custom URL **OR** my remote IP.
6. Use SSL is checked.
7. The server verifies successfully, and I click "Save".
8. The PMS server URL is retrieved as `https://mynetwork.mydomain.com:port`.

I think this is because both my local IP and my custom URL connections are listed as `local="1"`. So [this line in `plextv.get_real_pms_url()`](https://github.com/drzoidberg33/plexpy/blob/dev/plexpy/plextv.py#L89) grabs the connection with my custom URL instead of my IP addres.

---

> If you're trying to use a remote address when there is a private address available it won't be returned from the discover method due to my change at https://github.com/drzoidberg33/plexpy/pull/529/files#diff-cb42729bf12c3bdce109028c021370b7R501
> 
> Not sure why you would want to do this in the first place?

I am testing this from the settings page, not the setup wizard.
1. In the Plex server settings > Network > Secure connections is set to "Required".
2. I have remote access enabled.
3. In PlexPy > Settings > Plex Media Server, I enter my remote IP address.
4. Remote Server is checked.
5. Use SSL is checked.
6. The server does not verify, and I cannot click "Save".

---

Here is the full table:

| Secure Settings | Use SSL | Remote Server | IP/Hostname | Verifies | Connects | PMS URL |
| --- | --- | --- | --- | --- | --- | --- |
| Preferred | Unchecked | Unchecked | Local IP | Yes | Yes | http://192.168.xxx.xxx:32400 |
| Preferred | Unchecked | Unchecked | Custom URL | Yes | Yes | http://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Unchecked | Unchecked | Remote IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Local IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Custom URL | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Unchecked | Checked | Remote IP | Yes | Yes | http://174.xxx.xxx.xxx:xxxxx |
| Preferred | Checked | Unchecked | Local IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Unchecked | Custom URL | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Unchecked | Remote IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Preferred | Checked | Checked | Local IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Preferred | Checked | Checked | Custom URL | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Preferred | Checked | Checked | Remote IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Required | Unchecked | Unchecked | Local IP | Yes | No | http://192.168.xxx.xxx:32400 |
| Required | Unchecked | Unchecked | Custom URL | Yes | No | http://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Required | Unchecked | Unchecked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Unchecked | Checked | Local IP | Yes | No | http://174.xxx.xxx.xxx:xxxxx |
| Required | Unchecked | Checked | Custom URL | Yes | No | http://174.xxx.xxx.xxx:xxxxx |
| Required | Unchecked | Checked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Unchecked | Local IP | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Required | Checked | Unchecked | Custom URL | Yes | No | https://xxxxxxxxxx.xxxxxxxxxx.com:xxxxx |
| Required | Checked | Unchecked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |
| Required | Checked | Checked | Local IP | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Required | Checked | Checked | Custom URL | Yes | Yes | https://174-xxx-xxx-xxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:xxxxx |
| Required | Checked | Checked | Remote IP | No |  | Failed to access uri endpoint /identity. Is your server maybe accepting SSL connections only? '' |

**Comment by drzoidberg33** [comment]
Interesting. I don't use custom URLs in Plex so this isn't something I tested.

Does your custom URL resolve to a public IP?

I'm assuming so. The quickest fix I can think of is to just fetch the `plex.direct` one always.

**Comment by JonnyWong16** [comment]
> Does your custom URL resolve to a public IP?

Custom URL resolves to the same remote IP according to http://tracert.com/resolver.

And the new helper resolves correctly:

```
DEBUG :: CP Server Thread-11 : IP Checker :: Resolved mynetwork.mydomain.com to 174.xxx.xxx.xxx.
```

EDIT: Technically I don't need to set a custom URL in Plex, as the port is forwarded on my router, so it works either way.

**Comment by JonnyWong16** [comment]
Out of all this, I assume `localhost` or `127.0.0.1` will still work. I haven't tested that.

**Comment by drzoidberg33** [comment]
> Out of all this, I assume localhost or 127.0.0.1 will still work. I haven't tested that.

Yes, that still works.

Try the hacky fix I just committed and see if it fixes your issue.

**Comment by JonnyWong16** [comment]
> Try the hacky fix I just committed and see if it fixes your issue.

Yep, it grabs the correct `pms_url` based on Remote Server/Use SSL.

Only thing is still "Required" and entering my remote IP in PlexPy.

**Comment by JonnyWong16** [comment]
Not sure if this is just me. Websockets error with secure settings set to "Required"

```
Traceback (most recent call last):
  File ".\plexpy\logger.py", line 221, in new_run
    old_run(*args, **kwargs)
  File "C:\Python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File ".\plexpy\web_socket.py", line 56, in run
    ws = create_connection(uri)
  File "F:\Users\Jonny\Documents\GitHub\plexpy\lib/websocket\_core.py", line 104, in create_connection
    websock.connect(url, **options)
  File "F:\Users\Jonny\Documents\GitHub\plexpy\lib/websocket\_core.py", line 269, in connect
    self.handshake_response = handshake(self.sock, *addrs, **options)
  File "F:\Users\Jonny\Documents\GitHub\plexpy\lib/websocket\_handshake.py", line 58, in handshake
    status, resp = _get_resp_headers(sock)
  File "F:\Users\Jonny\Documents\GitHub\plexpy\lib/websocket\_handshake.py", line 109, in _get_resp_headers
    status, resp_headers = read_headers(sock)
  File "F:\Users\Jonny\Documents\GitHub\plexpy\lib/websocket\_http.py", line 196, in read_headers
    line = recv_line(sock)
  File "F:\Users\Jonny\Documents\GitHub\plexpy\lib/websocket\_socket.py", line 97, in recv_line
    c = recv(sock, 1)
  File "F:\Users\Jonny\Documents\GitHub\plexpy\lib/websocket\_socket.py", line 89, in recv
    raise WebSocketConnectionClosedException("Connection is already closed.")
websocket._exceptions.WebSocketConnectionClosedException: Connection is already closed.
```

**Comment by drzoidberg33** [comment]
Yep, secure websockets have never been supported. I'll see if I can't get that working while I'm at it.

**Comment by drzoidberg33** [comment]
Secure websockets fix pushed, seems to be working.

> Only thing is still "Required" and entering my remote IP in PlexPy.

I've removed public IPs being reported for local servers by the `discover` method as I don't see any reason to use the public IP when we have the servers local private IP.

So this would be expected then because you won't be able to retrieve `/identity` without the correct URL.

**Comment by JonnyWong16** [comment]
Secure websocket seems to be working.

> So this would be expected then because you won't be able to retrieve /identity without the correct URL.

Got it. Just tested with my VPN. Only the remote IP shows up in setup wizard and only the remote IP can be verified in the settings.

Everything seems good (except for the `PMS_SSL` in the config file).

**Comment by drzoidberg33** [comment]
> Got it. Just tested with my VPN. Only the remote IP shows up in setup wizard and only the remote IP can be verified in the settings.

I assume in this case this would be the correct behaviour?

Thanks for helping test this, any last gripes before I push to dev?

**Comment by JonnyWong16** [comment]
> I assume in this case this would be the correct behaviour?

Yep.
- Set up PlexPy on local network: only local IP in setup wizard and cannot verify remote IP in settings.
- Set up PlexPy on remote network: only remote IP in setup wizard and cannot verify local IP in settings.

Should be good to merge.

**Comment by drzoidberg33** [comment]
Thanks for the help!
